src/api/hooks/useUpdateRegistrationStatus.ts

```diff
@@ -0,0 +1,38 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setNotification } from '../../redux/notificationSlice';
+import { TerminatePublicationRequest, UnpublishPublicationRequest } from '../../types/registration.types';
+import { updateRegistrationStatus } from '../registrationApi';
+
+interface UpdateRequest {
+  registrationIdentifier: string;
+  data: UnpublishPublicationRequest | TerminatePublicationRequest;
+  onSuccess?: () => void;
+}
+
+export const useUpdateRegistrationStatus = () => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: ({ registrationIdentifier, data }: UpdateRequest) =>
+      updateRegistrationStatus(registrationIdentifier, data),
+    onSuccess: (response, variables) => {
+      dispatch(setNotification({ message: t('feedback.success.update_registration'), variant: 'success' }));
+      if (response.data) {
+        const key1 = ['registration', response.data.identifier, true];
```

Comment:
setter data automatisk fra responsen her når det går bra, sånn at vi unngår ekstra oppslag.
Dessverre har vi en litt rar hack med en `shouldNotRedirect=true|false` ved avpublisering, som gjør at vi må sørge for at data overskrives i begge tilfeller som gjør at det blir seende litt bloated ut.. :/